### DISCUSSION TITLE
Use own compiled leaf.a(v0.2.9) start the program, found that use multiple memory footprint than the official under occupation 10M

### DISCUSSION THREAD

**Discussion by liuxc0116** [eycorsican/leaf#68] [Q&A]
# Use own compiled leaf.a(v0.2.9) start the program, found that use multiple memory footprint than the official under occupation 10M

## Use the official version
![image](https://user-images.githubusercontent.com/6480317/108681061-96db5e80-7529-11eb-9dc9-08737a1092d6.png)

## Use own compiled version（Using `make ios` command to generate static library）
![image](https://user-images.githubusercontent.com/6480317/108681182-c7bb9380-7529-11eb-99c0-616514b26fe1.png)

From the debug log, we can see that the static library compiled by ourselves uses 10m more memory than the static library downloaded from the official by using the same code of version v0.2.9

I don't know what kind of situation would make such a difference

**Comment by eycorsican** [comment]
编译过程和环境你可以自己看github workflow，要是你保证不了相同编译环境，你也可以自己在github 上编译。

我自己本地编译没发现什么显著差异。

**Reply by liuxc0116** [reply]
谢谢，我试试。

**Reply by liuxc0116** [reply]
按照workflow里的步骤重新安装编译环境后，编译出来的包就正常了
```
curl https://sh.rustup.rs -o rustup-init.sh
bash rustup-init.sh -y --default-toolchain none --profile=minimal

source $HOME/.cargo/env
rustup toolchain install nightly
rustup default nightly

rustup target add aarch64-apple-ios
cargo install cargo-lipo
cargo install cbindgen
brew update && brew install llvm unzip protobuf

make ios
```
谢谢
